### ISSUE TITLE
`clone/get` should support setting annex.private=true

### ISSUE THREAD

**Issue by mih** [datalad/datalad#6456]
This was already requested some time ago in https://github.com/datalad/datalad/issues/5835 in a slightly different context, but it has never been done. Being able to do it is a major improvement when updating large datasets where repeated temporary clones need to be made to fit in the storage allotment.

In the simplest case the `--reckless` switch gets a dedicated mode for this.

**Comment by matrss** [comment]
For `clone` this is possible already, by passing additional options to git clone: `datalad clone <url> -c annex.private=true`.

This would also be useful for `create`, e.g. when setting up a mostly read-only dataset in a temporary location.
Appending additional options to the git command does not work here, because they are added at the wrong position.

**Comment by mih** [comment]
> This would also be useful for `create`, e.g. when setting up a mostly read-only dataset in a temporary location. Appending additional options to the git command does not work here, because they are added at the wrong position.

It seems to be doable already too:

```
% datalad -c annex.private=true create dummy
% cat dummy/.git/config
...
[annex]
        uuid = f1eb3d37-0251-4510-bc98-ad2a7b2fcb55
        version = 8
        private = true
...
```

However, this way of passing the config item does not work for `clone`. The resulting heterogeneity is undesirable.

**Comment by mslw** [comment]
Though at first glance the above doesn't fully work (trying on DataLad 0.17.10):

```
❱ datalad -c annex.private=true create dummy
❱ cd dummy 
❱ cat .git/config
...
[annex]
	uuid = 1245a968-4d01-4280-be8d-cc2cf66e9c96
	version = 10
	private = true
...
❱ git cat-file blob git-annex:uuid.log
1245a968-4d01-4280-be8d-cc2cf66e9c96 mszczepanik@bnbnb64:~/dummy timestamp=1673436462.930674266s
❱ ls .git/annex/journal-private
ls: cannot access '.git/annex/journal-private': No such file or directory
```
Can't see why, could that be due to the internal order of things within `create`; or did I simply make a mistake?

I am finishing a PR (above) in which I propose introducing `--reckless private` for `clone` & `get` and introducing a `--private` option for create. We can of course disciss whether that's an appropriate naming/interface.

**Comment by mih** [comment]
Yes, I can confirm too that I concluded to quickly. The config is set, but too late to have an impact on `git annex init`.
